### PR TITLE
Add Gas Price Data Source

### PR DESCRIPTION
**Pull request by augustbleeds**
This is in the same vein of work as https://github.com/smartcontractkit/chainlink-common/pull/509. 

Introducing a new gas price data source.

Merge Order

1. https://github.com/smartcontractkit/chainlink-common/pull/509
2. This PR
3. https://github.com/smartcontractkit/chainlink/pull/13200

### CODE REVIEW

File: go.mod
Comment (augustbleeds) [inline go.mod:5]: todo: remove once merged

File: go.mod
Comment (augustbleeds) [inline go.mod:9]: todo: change once merged

File: go.mod
Comment (augustbleeds) [inline go.mod:10]: todo: change once merged

File: median/aggregated_attribute_observation.go
Comment (augustbleeds) [inline median/aggregated_attribute_observation.go:37]: what i mean here is that if in the nil case, the `aggregated.GasPriceSubunit` will be set to nil because you can compare two nils (returns 0) and the median entry will be nil since all the gas data points are nil

Comment (augustbleeds) [reply median/aggregated_attribute_observation.go:37]: I thought about adding a special error case for if you have a nil value and a non-nil value -- but decided not to. It could already happen with juelsperfeecoin or the ocr value and we just trust it to panic in that case.

Comment (augustbleeds) [reply median/aggregated_attribute_observation.go:37]: EDIT: correction, it is currently impossible to have nil observation values. But with https://github.com/smartcontractkit/offchain-reporting/pull/412/files#diff-f47f225a9c371e457320c9da5a88b0927ce8060ce4c02aa8ba40bf92f010cf74R347 only GasPriceSubunit would be possible to be nil.

Thinking through the edge cases,
Is it better to 
a. fail if gas prices are a mixture of nil and non-nil? (this is what currently happens)
b. transmit the median value

Worst case for scenario (a) is that transmit() is blocked until all nops agree on nil vs non-nil. This would affect all chains, including EVM chains if some nops are incorrectly set.

Worst case scenario for (b) is that only Starknet would be blocked if the median value is `nil` because for EVM chains the value of GasPrice would never be used regardless of whether a nil or non-nil is the final value.

I think it's safer to go with (b)? If so, I can adjust the code

Comment (augustbleeds) [reply median/aggregated_attribute_observation.go:37]: EDIT: fixed so that nil values are never allowed for gas price (like for julesperfeecoin and datasourcevalue). so all values will technically panic on nil, but the data source never submits a nil.

File: median/aggregated_attribute_observation.go
Comment (augustbleeds) [inline median/aggregated_attribute_observation.go:37]: TODO: remove this ?

Comment (augustbleeds) [reply median/aggregated_attribute_observation.go:37]: we can add this but it's not necessary for now

File: go.mod
Comment (augustbleeds) [inline go.mod:7]: will point to master once https://github.com/smartcontractkit/chainlink-common/pull/509 is merged

File: median/aggregated_attribute_observation.go
Comment (augustbleeds) [inline median/aggregated_attribute_observation.go:37]: extracts the GasPriceSubunits to the aggregated observation. If it uses the default codec or codecs that do not know about this field, then field is ignored (see https://chainlink-core.slack.com/archives/C03EZMT3XPX/p1715627908908079?thread_ts=1711719895.085089&cid=C03EZMT3XPX). 

If it's on Starknet, the Starknet codec will utilize it

File: median/plugin.go
Comment (jmank88) [reply median/plugin.go:46]: Oh, or drop this var and just inline it down below instead?

**Review by jmank88 — DISMISSED** [review]
LGTM after common update

**Comment by augustbleeds** [comment]
> LGTM after common update

Thanks, will update now that chainlink-common has been merged here https://github.com/smartcontractkit/chainlink-common/pull/509

**Review by jmank88 — APPROVED** [review]

**Pull request merged**
